code/modules/asset_cache/transports/asset_transport.dm

```diff
@@ -0,0 +1,154 @@
+/// When sending mutiple assets, how many before we give the client a quaint little sending resources message
+#define ASSET_CACHE_TELL_CLIENT_AMOUNT 8
+
+/// Base browse_rsc asset transport
+/datum/asset_transport
+	var/name = "Simple browse_rsc asset transport"
+	var/static/list/preload
+	/// Don't mutate the filename of assets when sending via browse_rsc.
+	/// This is to make it easier to debug issues with assets, and allow server operators to bypass issues that make it to production.
+	/// If turning this on fixes asset issues, something isn't using get_asset_url and the asset isn't marked legacy, fix one of those.
+	var/dont_mutate_filenames = FALSE
+
+/// Called when the transport is loaded by the config controller, not called on the default transport unless it gets loaded by a config change.
+/datum/asset_transport/proc/Load()
+	// if (CONFIG_GET(flag/asset_simple_preload))
+	for(var/client/C in clients)
+		addtimer(CALLBACK(src, .proc/send_assets_slow, C, preload), 1 SECONDS)
+
+/// Initialize - Called when SSassets initializes.
+/datum/asset_transport/proc/Initialize(list/assets)
+	preload = assets.Copy()
+	// if (!CONFIG_GET(flag/asset_simple_preload))
+	// 	return
+	// for(var/client/C in GLOB.clients)
+	// 	addtimer(CALLBACK(src, .proc/send_assets_slow, C, preload), 1 SECONDS)
+
+
+/// Register a browser asset with the asset cache system
+/// asset_name - the identifier of the asset
+/// asset - the actual asset file (or an asset_cache_item datum)
+/// returns a /datum/asset_cache_item.
+/// mutiple calls to register the same asset under the same asset_name return the same datum
+/datum/asset_transport/proc/register_asset(asset_name, asset)
+	var/datum/asset_cache_item/ACI = asset
+	if (!istype(ACI))
+		ACI = new(asset_name, asset)
+		if (!ACI || !ACI.hash)
+			CRASH("ERROR: Invalid asset: [asset_name]:[asset]:[ACI]")
+	if (SSassets.cache[asset_name])
+		var/datum/asset_cache_item/OACI = SSassets.cache[asset_name]
+		OACI.legacy = ACI.legacy = (ACI.legacy|OACI.legacy)
+		OACI.namespace_parent = ACI.namespace_parent = (ACI.namespace_parent | OACI.namespace_parent)
+		OACI.namespace = OACI.namespace || ACI.namespace
+		if (OACI.hash != ACI.hash)
+			var/error_msg = "ERROR: new asset added to the asset cache with the same name as another asset: [asset_name] existing asset hash: [OACI.hash] new asset hash:[ACI.hash]"
```

Comment:
/*var/error_msg = "ERROR: new asset added to the asset cache with the same name as another asset: [asset_name] existing asset hash: [OACI.hash] new asset hash:[ACI.hash]" // commented out because 800 MB logs crash PCs
			log_asset(error_msg) */
			return TRUE